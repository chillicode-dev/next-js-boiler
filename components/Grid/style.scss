.container {
  max-width: #{$containerMaxWidth}px;
  margin: 0 auto;
}

.row {
  display: flex;
  flex-wrap: wrap;
}

.column {
  width: span(2);
  margin-right: gutter();
  margin-bottom: gutter();
  background-color: gray;
  outline: 1px solid red;
  @include disableLastMargin(right);
  @include disableNthMargin(6n, right);
}

.input {
  @include placeholder(red);
}
