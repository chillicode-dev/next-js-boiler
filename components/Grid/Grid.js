import React from 'react';
import style from './style.scss';

const Grid = () => {
  return (
    <div className={style.container}>
      <div className={style.row}>
        <div className={style.column}>1</div>
        <div className={style.column}>2</div>
        <div className={style.column}>3</div>
        <div className={style.column}>4</div>
        <div className={style.column}>5</div>
        <div className={style.column}>6</div>
        <div className={style.column}>7</div>
        <div className={style.column}>8</div>
        <div className={style.column}>9</div>
        <div className={style.column}>10</div>
        <div className={style.column}>11</div>
      </div>
    </div>
  );
};

export default Grid;
