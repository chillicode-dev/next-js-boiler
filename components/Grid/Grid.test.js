import React from 'react';
import renderer from 'react-test-renderer';
import Grid from './Grid';

it('Renders Grid correctly', () => {
  const tree = renderer.create(<Grid />).toJSON();
  expect(tree).toMatchSnapshot();
});
