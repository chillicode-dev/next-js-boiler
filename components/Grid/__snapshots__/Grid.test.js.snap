// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Grid correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="column"
    >
      1
    </div>
    <div
      className="column"
    >
      2
    </div>
    <div
      className="column"
    >
      3
    </div>
    <div
      className="column"
    >
      4
    </div>
    <div
      className="column"
    >
      5
    </div>
    <div
      className="column"
    >
      6
    </div>
    <div
      className="column"
    >
      7
    </div>
    <div
      className="column"
    >
      8
    </div>
    <div
      className="column"
    >
      9
    </div>
    <div
      className="column"
    >
      10
    </div>
    <div
      className="column"
    >
      11
    </div>
  </div>
</div>
`;
